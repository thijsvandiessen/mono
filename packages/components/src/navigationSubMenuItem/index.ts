export { NavigationSubMenuItem } from './navigationSubMenuItem.jsx'
