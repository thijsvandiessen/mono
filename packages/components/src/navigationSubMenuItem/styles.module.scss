.link {
  display: block;
  color: var(--colour-primary-dark);
  font-family: var(--typography-serif);
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;

  &:hover,
  &:focus {
    text-decoration: none;
    color: var(--colour-secondary);
  }
}
