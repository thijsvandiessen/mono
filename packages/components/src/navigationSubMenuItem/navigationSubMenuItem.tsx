import Link from 'next/link'
import React from 'react'
import classNames from 'classnames'
import { slugFormatter } from '@mono/utils/src/slugFormatter'
import styles from './styles.module.scss'

export interface NavigationSubMenuItemProps {
  slug?: string | null
  label?: string | null
  onClick: () => void
}

export const NavigationSubMenuItem = ({
  slug,
  label,
  onClick,
}: NavigationSubMenuItemProps) => {
  if (!slug)
    return <li className={classNames(styles.navItem)}>No link specified</li>
  if (!label)
    return <li className={classNames(styles.navItem)}>No label specified</li>

  return (
    <li className={classNames(styles.navItem)}>
      <Link
        className={classNames(styles.link, 'text-large')}
        href={slugFormatter({ slug })}
        onClick={onClick}
      >
        {label}
      </Link>
    </li>
  )
}
