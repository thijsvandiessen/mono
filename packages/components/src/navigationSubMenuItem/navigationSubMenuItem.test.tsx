import { describe, expect, it } from 'vitest'
import { render, screen } from '@testing-library/react'
import { NavigationSubMenuItem } from '../navigationSubMenuItem'
import React from 'react'

describe('NavigationSubMenuItem', () => {
  it('should render a link', () => {
    const { container } = render(
      <NavigationSubMenuItem
        onClick={() => null}
        slug={'url'}
        label={'homepage'}
      />
    )

    expect(screen.getByRole('link', { name: 'homepage' }))
    expect(container).toMatchSnapshot()
  })

  it('should render a link warning', () => {
    render(<NavigationSubMenuItem onClick={() => null} />)

    expect(screen.getByText('No link specified'))
  })

  it('should render a label warning', () => {
    render(<NavigationSubMenuItem onClick={() => null} slug={'test'} />)

    expect(screen.getByText('No label specified'))
  })
})
