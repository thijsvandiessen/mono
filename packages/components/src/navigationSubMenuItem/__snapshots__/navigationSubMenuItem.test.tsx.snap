// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavigationSubMenuItem > should render a link 1`] = `
<div>
  <li
    class="_navItem_16c685"
  >
    <a
      class="_link_16c685 text-large"
      href="/url"
    >
      homepage
    </a>
  </li>
</div>
`;
